src/fileformats/ocd_file_export.cpp

```diff
@@ -1949,9 +1949,15 @@ void OcdFileExport::setupTextSymbolSpecial(const TextSymbol* text_symbol, OcdTex
 	ocd_text_special.line_below_offset = decltype(ocd_text_special.line_below_offset)(convertSize(qRound(1000 * text_symbol->getLineBelowDistance())));
 	
 	ocd_text_special.num_tabs = text_symbol->getNumCustomTabs();
-	auto last_tab = std::min(ocd_text_special.num_tabs, decltype(ocd_text_special.num_tabs)(std::extent<typename std::remove_pointer<decltype(ocd_text_special.tab_pos)>::type>::value));
-	for (auto i = 0u; i < last_tab; ++i)
-		ocd_text_special.tab_pos[i] = convertSize(text_symbol->getCustomTab(i));
+	auto max_tabs = decltype(ocd_text_special.num_tabs)(std::extent<typename std::remove_pointer<decltype(ocd_text_special.tab_pos)>::type>::value);
+	if (ocd_text_special.num_tabs > max_tabs)
+	{
+		ocd_text_special.num_tabs = max_tabs;
+		addWarning(::OpenOrienteering::OcdFileExport::tr("In text symbol %1: exporting only %2 custom tabulator positions")
+		           .arg(text_symbol->getPlainTextName()).arg(max_tabs));
+	}
+	for (auto i = 0u; i < ocd_text_special.num_tabs; ++i)
+		ocd_text_special.tab_pos[i] = quint32((text_symbol->getCustomTab(i) + 5) / 10);
```

Comment:
Two comments on the change. First, the trivial one: every commit should change just one thing. Even if that's a single-line change, or even a single character. The presence of the phrase "in addition" in the commit message suggests that the commit contains multiple changes.

The first change is the `num_tabs` trimming. The second one is the `convertSize()` overload selection problem. I'm fine with the first change but the second tries to walk around a can of worms. I understand the motivation but when we are at it, the code should be cleaned up.

The trouble with `convertSize()` is multifaceted. Let me list my findings:
1) In the file format documentation, the tab positions are [LongInt](http://web.archive.org/web/20200728124010/http://docwiki.embarcadero.com/Libraries/Sydney/en/System.Longint) type. It corresponds to `qint32` in the Qt world. But the Mapper definition is an unsigned integer at `src/fileformats/ocd_types_v8.h:334`. That is the smaller issue.
2) `convertSize()` has two overloads and both of them convert the data type to a smaller one. I don't get the point. I'd expect the data type size to remain the same after division by 10. In any case, triggering the selection of the `qint64 -> qint32` function variant is a better course of action than open-coding the conversion.

My recommendation is, therefore, to split the patch into two - one with the addition of warning and the second addressing the type mess. HTH

Reply:
Thank you for your review and proposals Libor.

I already stumbled across the issue with longint being mapped to quint32 in the ocd_types_v8.h (which does not matter due to the actual value range in this special case).
However this affects more mappings like LineSpace, IndentFirst or IndentOther which are SmallInt but mapped to quint16.

The convertSize() has two overloads but the convertSize(qint64 size) seems not to used at all.
Since the conversion from quint32 to qint32 is only needed once it seemed to me to be the easiest way to directly convert it.
AFAIK it's not possible to overload qint16 convertSize(qint32 size) by qint32 convertSize(qint32 size), since the input parameter/type pattern must be different. I don't know right know whether a convertSize((int64)..) cast would solve it, but that looks somehow weird.
OcdFileImport uses class functions and templates for the convertOcdString functionality. Would that be a path to follow?

Reply:
My preference would be towards a function template as well, Matthias. And at the exact spot where we need 32-bit precision, I'd go with explicit instantiation (`convertSize<quint32>(text_symbol->getCustomTab(i))`).

However, @dg0yt can have a much better idea about how to eliminate the narrowing type conversion. Kai, any comments on this topic, please?